### PR TITLE
Moving login test in a commonTests directory

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Adding commonTests directory with login test as an example, and create PageObjects var to be detected in mocha context
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="sanity/02_productsBO/01_filterProducts" URL_FO=SHOP_URL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15630)

### CODE REVIEW

File: tests/puppeteer/.eslintrc.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ module.exports = {
   },
   rules: {
     'no-plusplus': [2, { allowForLoopAfterthoughts: true }],
+    "func-names": "off",
```
Comment (SimonGrn) [inline tests/puppeteer/.eslintrc.js:20]: THANK YOU !

File: tests/puppeteer/campaigns/commonTests/loginBO.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const {expect} = require('chai');
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/commonTests/loginBO.js:1]: Does this works if it's outside of the export ?

**Review by SimonGrn — DISMISSED** [review]
Thanks

File: tests/puppeteer/campaigns/commonTests/loginBO.js (hunk 2 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/commonTests/loginBO.js:1]: yes

File: tests/puppeteer/campaigns/upgrade/upgradeShop.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
+// Using chai
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/upgrade/upgradeShop.js:1]: Not necessary I guess ?

File: tests/puppeteer/campaigns/utils/helpers.js [hunk 4]
```diff
@@ -15,4 +15,10 @@ module.exports = {
       },
     });
   },
+  async newTab(browser) {
+    return browser.newPage();
+  },
+  async closeBrowser(browser) {
+    return browser.close();
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/utils/helpers.js:22]: 👍

**Review by SimonGrn — APPROVED** [review]
Thanks

File: tests/puppeteer/campaigns/upgrade/upgradeShop.js (hunk 3 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/upgrade/upgradeShop.js:1]: Comment was not in the right place, I just moved it

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**